### PR TITLE
Add explicit port number support in vds URL

### LINKED ISSUE #133: Allow explicit ports number in storage account url
The following to blob store urls are identical:

`https://<name>.blob.core.windows.net` and `https://<name>.blob.core.windows.net:443` are identical. Currently our allow list check will deny the later, as `:443` is included in the string compare. We should remove the port from the comparison

### PR DESCRIPTION
**Pull request by AdilKhurshid**
closes #133

### CODE REVIEW

File: api/request.go [hunk 1]
```diff
@@ -41,16 +42,23 @@ type Normalizable interface {
 }
 
 func (r *RequestedResource) NormalizeConnection() error {
-	blob := strings.Split(strings.TrimSpace(r.Vds), "?")
-	r.Vds = blob[0]
+	url, err := url.Parse(r.Vds)
+	if  err  != nil {
+		return core.NewInvalidArgument(err.Error())
+	}
+	if url.Hostname() != "" && url.Scheme != "" {
```
Comment (achaikou) [inline api/request.go:49]: Maybe we can do the same logic a bit more straighforward? Allow the library to gather the string back

```
	if strings.TrimSpace(r.Sas) == "" {
		if url.RawQuery == "" {
		    return core.NewInvalidArgument("No valid Sas token is found in the request")
		}
		r.Sas = url.RawQuery
	}

	url.RawQuery = ""
	url.Host = url.Hostname()
	r.Vds = url.String()
```

Now it is not clear why we need to check hostname and scheme for emptiness.

File: api/request.go [hunk 2]
```diff
@@ -41,16 +42,23 @@ type Normalizable interface {
...
+	} else {
+		r.Vds = url.Path
+	}
+
 	if strings.TrimSpace(r.Sas) != "" {
 		return nil
 	}
-
-	if len(blob) < 2 {
+	if strings.TrimSpace(r.Sas) == "" && url.RawQuery == "" {
```
[the remark this answers is not in the document]
Comment (AdilKhurshid) [reply api/request.go:58]: yes, you are right here. I will change this logic

File: api/request_test.go [hunk 3]
```diff
@@ -237,3 +237,35 @@ func TestExtractSasFromUrl(t *testing.T) {
 		}
 	}
 }
+
+func TestPortPresenceInURL(t *testing.T) {
+
+	testCases := []struct {
+		request     RequestedResource
+		expected    string
+		shouldError bool
```
[the remark this answers is not in the document]
Comment (AdilKhurshid) [reply api/request_test.go:246]: I don't think there is any other case that can cause an error. I will remove that parameter. Thanks

File: api/request.go (hunk 1 above)
Comment (AdilKhurshid) [reply api/request.go:49]: I will replace the code with the logic you suggested that is better 😊

**Review by achaikou — COMMENTED** [review]
Looks good to me :+1:  !
Rebase, fixup and I'll approve.

**Review by achaikou — APPROVED** [review]

**Pull request merged**